tests/Unit/Elliptic/BoundaryConditions/Test_BoundaryConditionType.cpp

```diff
@@ -0,0 +1,23 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <string>
+
+#include "Elliptic/BoundaryConditions/BoundaryConditionType.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Utilities/GetOutput.hpp"
+
+namespace elliptic {
+
+SPECTRE_TEST_CASE("Unit.Elliptic.BoundaryConditionType", "[Unit][Elliptic]") {
+  CHECK(get_output(BoundaryConditionType::Dirichlet) == "Dirichlet");
+  CHECK(get_output(BoundaryConditionType::Neumann) == "Neumann");
+  CHECK(TestHelpers::test_creation<BoundaryConditionType>("Dirichlet") ==
+        BoundaryConditionType::Dirichlet);
+  CHECK(TestHelpers::test_creation<BoundaryConditionType>("Neumann") ==
+        BoundaryConditionType::Neumann);
+}
```

Comment:
[optional] Could add a test that option-parsing correctly fails if you pass in an invalid string